qutebrowser/mainwindow/tabwidget.py

```diff
@@ -457,12 +450,35 @@ def minimumTabSizeHint(self, index):
             padding_h += self.style().pixelMetric(
                 PixelMetrics.icon_padding, None, self)
         height = self.fontMetrics().height() + padding_v
-        width = (self.fontMetrics().width('\u2026') + icon_size.width() +
+        width = (self.fontMetrics().width(text) + icon_size.width() +
                  padding_h + config.val.tabs.width.indicator)
         return QSize(width, height)
 
-    def tabSizeHint(self, index):
-        """Override tabSizeHint so all tabs are the same size.
+    def _tabTotalWidthPinned(self):
```

Comment:
All those method names should be `_snake_case` as per [pep8](http://pep8.org/) (only Qt overrides are camelCase).